newrelic-agent/src/main/java/com/newrelic/agent/config/ConfigServiceFactory.java

```diff
@@ -25,16 +25,19 @@ public static ConfigService createConfigService(AgentConfig config, Map<String,
 
     @VisibleForTesting
     public static ConfigService createConfigServiceUsingSettings(Map<String, Object> settings) {
-        return new ConfigServiceImpl(AgentConfigImpl.createAgentConfig(settings), null, settings, false);
+        Map<String, Object> deObfuscatedSettings = new ObscuringConfig(
+                settings, AgentConfigImpl.SYSTEM_PROPERTY_ROOT).getDeobscuredProperties();
+        return new ConfigServiceImpl(AgentConfigImpl.createAgentConfig(settings), null, deObfuscatedSettings, false);
     }
 
     public static ConfigService createConfigService(Logger log, boolean checkConfig) throws ConfigurationException, ForceDisconnectException {
         File configFile = getConfigFile(log);
         Map<String, Object> configSettings = getConfigurationFileSettings(configFile, log);
-
-        AgentConfig config = AgentConfigImpl.createAgentConfig(configSettings);
+        Map<String, Object> deObfuscatedSettings = new ObscuringConfig(
```

Comment:
ObscuringConfig deobfuscates and it is move here, earlier in the lifecycle, so that deObfuscated settings are into `ConfigServiceImpl`